libs/store/src/lib/domain/query/query.ts

```diff
@@ -0,0 +1,243 @@
+    _select: string[];
+
+    /** @hidden - stores current expand properties */
+    _expand: string[];
+
+    constructor(
+        private resultType: Type<TModel>,
+        private service: EntityCollectionService<TModel>,
+        private adapter: QueryAdapter<TModel>
+    ) {}
+
+    /**
+     * Replace current filter settings.
+     * @param predicate Predicate object which contains new filter criteria.
+     */
+    where(predicate: Predicate<TModel>): Query<TModel> {
+        this._predicate = predicate;
+        return this;
+    }
+
+    /**
+     * Set keyword for search term.
+     * @param keyword
+     */
+    keyword(keyword: string): Query<TModel> {
+        this._keyword = keyword;
+        return this;
+    }
+
+    /**
+     * Set order by rules for query.
+     * @param orderBys Set of OrderBy objects.
+     */
+    orderBy<TProperty extends keyof TModel> (...orderBys: Array<OrderBy<TModel, TProperty>> ): Query < TModel > {
+        this._orderByFields = orderBys;
+        return this;
+    }
+
+    /**
+     * Set first index of result set for paging.
+     * @param offset Index number of first result.
+     */
+    firstResult(offset: number): Query<TModel> {
+        this._offset = offset;
+        this._suppressPageReset = true;
+        return this;
+    }
+
+    /**
+     * Set page size for result set.
+     * @param pageSize Number of items returned per page
+     */
+    maxResults(pageSize: number): Query<TModel> {
+        this._pageSize = pageSize;
+        return this;
+    }
+
+    /**
+     * Suspends resetting of page index when query keyword, where, or
+     * order by clauses have been changed.
+     */
+    suppressPageReset(): Query<TModel> {
+        this._suppressPageReset = true;
+        return this;
+    }
+
+    /**
+     * Set flag to include "count" in response.
+     * If "true" the query string will include parameter "$count=true".
+     * If "false" the parameter will not be included.
+     * @param flag
+     */
+    includeCount(flag: boolean): Query<TModel> {
+        this._includeCount = flag;
+        return this;
+    }
+
+    /**
+     * Set list of select parameters. Select parameters are used to limit
+     * the enitity properties included in the return data.
+     * @param select List of properties to include in response data
+     */
+    select<TP extends keyof TModel> (...select: Array<TP> ): Query<TModel> {
+        this._select = select as string[];
+        return this;
+    }
+
+    /**
+     * Set list of expand parameters. Expand parameters are used to ask the server
+     * to include relational data.
+     * @param extend List of expanded properties to include in response data
+     */
+    expand<TP extends keyof TModel> (...expand: Array<TP> ): Query<TModel> {
+        this._expand = expand as string[];
+        return this;
+    }
+
+    /**
+     * Initiate query and return observable
+     */
+    fetch(): Observable<TModel | Array<TModel>> {
```

Comment:
Doesn't `this.service.getWithQuery(query)` always return entity collection?

Reply:
yes you're correct. It should just be `Array<TModel>>`